### PR TITLE
Add layers argument to support reloading and fix iteration upper bound.

### PR DESCRIPTION
**Pull request by allanzelener**
Layers passed explicitly to LSTM function to allow reloading batch norm layer parameters. Note that layers are not currently supported as arguments in optimize mode so this can only be used in direct mode. Layers also need to be switched between train/evaluate outside of the LSTM function.

Also fixed bug in iteration when making batch norm layers.

### CODE REVIEW

**Comment by alexbw** [comment]
This ready to be merged?

**Comment by allanzelener** [comment]
Yes, unless the nn.batchNormalization implementation changes this is as good as this version gets.

I'm working on an autograd native version that may resolve some of the issues here but there may be performance tradeoffs so that's a more significant PR.

File: src/model/RecurrentLSTMNetwork.lua [hunk 1]
```diff
@@ -12,9 +12,9 @@ return function(opt, params)
    local batchNormalization = opt.batchNormalization or false
    local maxBatchNormalizationLayers = opt.maxBatchNormalizationLayers or 10
 
-   -- container:
+   -- containers:
    params = params or {}
-   layers = layers or {}
```
Comment (alexbw) [inline src/model/RecurrentLSTMNetwork.lua]: Why do you remove the `layers or {}` line? I think some folks might depend on this (although I do not) -- I believe it lets folks concatenate the output of the recurrent net built in this file to a passed-in set of layers.

Comment (allanzelener) [inline src/model/RecurrentLSTMNetwork.lua]: This line was something I added in the original PR since I didn't really understand how the other models used layers yet. Since the LSTM is implemented in autograd it can't be concatenated with a sequence of nn modules so passing in preceding layers doesn't make sense. While the batch norm layers here share parameters, they are all disconnected and applied independently at each time step.

**Comment by alexbw** [comment]
lgtm, thanks for the contribution!
